bindings/python/src/session.cpp

```diff
@@ -413,6 +413,17 @@ namespace
         s.set_alert_notify(std::bind(&alert_notify, cb));
     }
 
+    void alert_fd_notify(int const fd)
+    {
+        std::uint8_t dummy = 0;
+        ::write(fd, &dummy, 1);
```

Comment:
Unfortunately, `signal.set_wakeup_fd` is somewhat complicated for a reason. Although on UNIX-like systems `write` is ok for sockets, on Windows, you have to use `send` for sockets.

Reply:
I see, so this patch doesn't support sockets on windows. Sockets on windows also aren't `int`s, they are `HANDLE`s. I wonder what `fileno()` returns on a socket in python on windows then..

Reply:
`HANDLE` is actually an integer, just even more opaque than fd on UNIX-like systems (the latter are guaranteed to be assigned sequentially, 0, 1 and 2 have predefined meaning, etc.).

Reply:
`HANDLE` is a void pointer, which is likely twice as wide as `int`

Reply:
https://github.com/python/cpython/blob/master/Modules/socketmodule.c#L3375
https://github.com/python/cpython/blob/master/Modules/socketmodule.h#L207

Looks like `socket.fileno()` is always a **python** `int`; on windows it's cast from `SOCKET`.

https://github.com/python/cpython/blob/master/Modules/_io/fileio.c#L382

For normal files python calls `_wopen` on windows, so you get file descriptor that fits in a C `int`.

https://github.com/python/cpython/blob/master/Modules/signalmodule.c#L670

Python's `signal.set_wakeup_fd` queries the fd with winsock, and if that fails, assumes it's a normal int file descriptor.